wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/AuroraHostListProvider.java

```diff
@@ -170,8 +171,10 @@ protected void init() throws SQLException {
           TimeUnit.MILLISECONDS.toNanos(CLUSTER_TOPOLOGY_REFRESH_RATE_MS.getInteger(properties));
       this.clusterInstanceTemplate =
           CLUSTER_INSTANCE_HOST_PATTERN.getString(this.properties) == null
-              ? new HostSpec(rdsHelper.getRdsInstanceHostPattern(originalUrl))
-              : new HostSpec(CLUSTER_INSTANCE_HOST_PATTERN.getString(this.properties));
+              ? this.hostListProviderService.getHostSpecBuilder()
```

Comment:
This piece is a bit unreadable. I'd recommend to define a local variable with host spec builder.